### ISSUE TITLE
SecretBuilder throws exception on a valid secret name

### ISSUE THREAD

**Issue by bobbydazzler** [CompositionalIT/farmer#252]
I think adding validation to SecretConfig.Create has introduced a regression via the SecretBuilder.Yield function.

```
let mySecret = secret {
    name "name-of-secret"
}

```
Unhandled exception. System.Exception: Key Vault key names must be a 1-127 character string, starting with a letter and containing only 0-9, a-z, A-Z, and -. '' is invalid.
   at Microsoft.FSharp.Core.PrintfModule.PrintFormatToStringThenFail@1637.Invoke(String message) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\printf.fs:line 1637
   at Farmer.Builders.KeyVault.SecretConfig.Create(String key)
   at Farmer.Builders.**KeyVault.SecretBuilder.Yield**(Unit _arg1)
   at <StartupCode$FarmerApp>.$Program.main@()

Builder.KeyVault.fs
```
type SecretBuilder() =
    member __.Yield (_:unit) = SecretConfig.Create "" // fails validation

```
type SecretConfig =
...
```
static member Create (key:string) =
        let charRulesPassed =
            let charRules = [ Char.IsLetterOrDigit; (=) '-' ]
            key |> Seq.forall(fun c -> charRules |> Seq.exists(fun r -> r c))
        let stringRulesPassed =
            [ (fun l -> String.length l <= 127)
              String.IsNullOrWhiteSpace >> not ]
            |> Seq.forall(fun r -> r key)

        if not (charRulesPassed && stringRulesPassed) then
            failwithf "Key Vault key names must be a 1-127 character string, starting with a letter and containing only 0-9, a-z, A-Z, and -. '%s' is invalid." key

```

**Comment by isaacabraham** [comment]
Doh! I'll fix this.

**Comment by isaacabraham** [comment]
In the short term you can use the `secret` keywords on the builder.

**Comment by isaacabraham** [comment]
Fixed by [this commit](https://github.com/CompositionalIT/farmer/commit/a9812a66d1405a394d36acd9b769d29510f5ad7b).

**Comment by isaacabraham** [comment]
@bobbydazzler Fixed in 0.20.1.
